astropy/nddata/utils.py

```diff
@@ -563,7 +564,8 @@ def __init__(self, data, position, size, wcs=None, mode='trim',
                     raise ValueError('shape can contain Quantities with only '
                                      'pixel or angular units')
 
-        data = np.asanyarray(data)
+        if not isinstance(data, Section):  # Accept lazy-loaded image sections
+            data = np.asanyarray(data)
```

Comment:
Note: this small change enables `Cutout2D` to use data obtained via `ImageHDU.section`, which is useful for reasons explained in this section of the docs: https://astropy--13238.org.readthedocs.build/en/13238/io/fits/usage/cloud.html#using-cutout2d-with-cloud-hosted-fits-files